shared/AppInsightsCore/src/JavaScriptSDK/BaseCore.ts

```diff
@@ -249,14 +265,182 @@ export class BaseCore implements IAppInsightsCore {
             };
 
             _self.releaseQueue = () => {
-                if (_eventQueue.length > 0) {
-                    arrForEach(_eventQueue, (event: ITelemetryItem) => {
+                if (_isInitialized && _eventQueue.length > 0) {
+                    let eventQueue = _eventQueue;
+                    _eventQueue = [];
+
+                    arrForEach(eventQueue, (event: ITelemetryItem) => {
                         _self.getProcessTelContext().processNext(event);
                     });
-
-                    _eventQueue = [];
                 }
             };
+
+            /**
+             * Periodically check logger.queue for log messages to be flushed
+             */
+             _self.pollInternalLogs = (eventName?: string): number => {
+                 _internalLogsEventName = eventName;
+
+                let interval = _self.config.diagnosticLogInterval;
+                if (!interval || !(interval > 0)) {
+                    interval = 10000;
+                }
+                if(_internalLogPoller) {
+                    clearInterval(_internalLogPoller);
+                }
+                _internalLogPoller = setInterval(() => {
+                    _flushInternalLogs();
+                }, interval) as any;
+
+                return _internalLogPoller;
+            }
+
+            /**
+             * Stop polling log messages from logger.queue
+             */
+            _self.stopPollingInternalLogs = (): void => {
+                if(_internalLogPoller) {
+                    clearInterval(_internalLogPoller);
+                    _internalLogPoller = 0;
+                    _flushInternalLogs();
+                }
+            }
+
+            // Add addTelemetryInitializer
+            proxyFunctions(_self, () => _telemetryInitializerPlugin, [ "addTelemetryInitializer" ]);
+
+            _self.getPlugin = _getPlugin;
+
+            function _initDefaults() {
```

Comment:
Just setting up a pattern so that during initial construction and after getting "unloaded" this function will be called this is to avoid having two places in the code and the potential side effects (bugs) thie might cause.